src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -36,6 +36,44 @@ public AnalyzeCommandBaseTests(ITestOutputHelper output)
             Output.WriteLine($"The seed that will be used is: {TestRule.s_seed}");
         }
 
+        [Fact]
+        public void AnalyzeCommandBase_ScanWithNoValidTargetsAndOneOrMoreSkippedFilesEmitsSkippedFilesWarning()
+        {
+            var logger = new TestMessageLogger();
+
+            var artifact = new EnumeratedArtifact(new FileSystem())
+            { 
+                Uri = new Uri(@"c:\testfile.txt"),
+                Contents = new string('x', 1025), // Just exceeds 1k.
+            };
+
+            var context = new TestAnalysisContext
+            {
+                TargetsProvider = new ArtifactProvider(new[] { artifact }),
+                MaxFileSizeInKilobytes = 1,
+                Logger = logger,
+            };
+
+            int result = new TestMultithreadedAnalyzeCommand().Run(options: null, ref context);
+
+            RuntimeConditions conditions = RuntimeConditions.NoValidAnalysisTargets |
+                                           RuntimeConditions.OneOrMoreFilesSkippedDueToExceedingSizeLimits;
+
+            context.RuntimeErrors.Should().Be(conditions);
+
+            logger.ConfigurationNotifications.Where(n => n.Level == FailureLevel.Note).Count().Should().Be(1);
+            Notification note = logger.ConfigurationNotifications.Where(n => n.Level == FailureLevel.Note).First();
+            note.Descriptor.Id.Should().Be(Notes.Msg002_FileExceedingSizeLimitSkipped);
+
+            logger.ConfigurationNotifications.Where(n => n.Level == FailureLevel.Warning).Count().Should().Be(1);
+            Notification warning = logger.ConfigurationNotifications.Where(n => n.Level == FailureLevel.Warning).First();
+            warning.Descriptor.Id.Should().Be(Warnings.Wrn997_OneOrMoreFilesSkippedDueToExceedingSizeLimits);
```

Comment:
>Should [](http://example.com/codeflow?start=34&length=6)

suggest a simple wrap so all failure can surface
using (new AssertionScope()){   }

Reply:
I always forget about this, THANK YOU.